File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        //sort karsakthe ho hamesha think out of the box its
        //really necesary dont forget

        Arrays.sort(strs);
        int ind=Math.min(strs[0].length(),strs[strs.length-1].length());
        String f=strs[0];
        String l=strs[strs.length-1];
        int i=0,c=0;;
        while(i<ind){
            if(f.charAt(i)==l.charAt(i)){
               c++;
               i++;
            }
            else break;
        }

        return f.substring(0,c);
    }
}

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 1:
            return strs[0]
        strs.sort()

        for i in range(len(strs[0])):
            if strs[0][i] != strs[-1][i]:
                return strs[0][:i]
        
        return strs[0]

        